generators/php/base/src/asIs/Client/RawClient.Template.php

```diff
@@ -178,17 +179,28 @@ private function encodeRequestBody(
     private function buildJsonBody(
         mixed $body,
         array $options,
-    ): mixed {
+    ): mixed
+    {
         $overrideProperties = $options['bodyProperties'] ?? [];
+        if(is_array($body) && self::isSequential($body)){
+            return array_merge($body, $overrideProperties);
+        }
+
         if ($body instanceof JsonSerializable) {
-            $serialized = $body->jsonSerialize();
-            return is_array($serialized)
-                ? array_merge($serialized, $overrideProperties)
-                : $serialized;
+            $result = $body->jsonSerialize();
         }
-        return is_array($body)
-            ? array_merge($body, $overrideProperties)
-            : $body;
+        else{
+            $result = $body;
+        }
+        if(is_array($result)){
+            $result = array_merge($result, $overrideProperties);
+            if(empty($result)){
+                // force to be serialized as {} instead of []
+                return (object)($result);
+            }
+        }
```

Comment:
```suggestion
        if (is_array($result)){
            $result = array_merge($result, $overrideProperties);
            if (empty($result)){
                // force to be serialized as {} instead of []
                return (object)($result);
            }
        }
```